scRNA-seq/01-normalizing_scRNA-seq.Rmd

```diff
@@ -78,84 +59,104 @@ sc_metadata <- readr::read_tsv(file.path("data", "unfiltered_darmanis_metadata.t
 overall. 
-To do this, let's calculate the gene means and make a density plot. 
+Use a command we have used previously to look at the first few rows of `sc_data`.
+
+```{r}
+# Put a command here that would allow you to see the first few rows of `sc_data`
+```
+
+You'll notice the first column contains the gene information and the rest are samples. 
+For the analyses in this notebook, the gene column (which is a `character` type), will get in 
+the way, so turn this column into the rownames instead (so it is out of our way but we still
+have the information).
+
+```{r Remove the gene column}
+# Set the column as the gene names
+sc_data <- sc_data %>% tibble::column_to_rownames("genes")
+```
+
+Let's look at how the gene means look in this dataset. 
+We will use `apply` in order to calculate things across our data.frame. 
+The second argument in `apply` specifies whether we are calculating by rows or columns. 
+(1 = rows, 2 = columns).
 
 ```{r}
 # Let's calculate the gene means and make a density plot
-gene_means <- apply(sc_data[, -1], 1, mean)
+gene_means <- apply(sc_data, 1, mean)
+# What do you think the dimensions of `gene_means` are? 
+```
 
-# Plot the means
-qplot(gene_means, geom = "density", xlab = "gene mean counts") + 
-  geom_density(fill = "lightblue") +
-  cowplot::theme_cowplot() 
+Let's make our first density plot with these data.
+
+```{r}
+# Plot the density of the means using ggplot2's `qplot`
+qplot(gene_means, geom = "density", xlab = "Gene Mean Counts") 
 ```
 
 The counts data has so many zeroes, which we should expect. 
 83.3% of the data points are zeroes. 
-This is not out of the ordinary for single-cell RNA-seq data. 
-This means our data consists of a few genes that are expressed a lot, and the 
-rest of the genes are barely detected. 
 
 ## Filter the counts matrix
   
-Due to the high risk of technical artifacts, we don't necessarily want 
-to keep the data for all the cells and genes. 
 The small amount of RNA in a single cell results in higher chances of errors or 
-biases in amplification steps for some genes. 
+biases in amplification steps, so we don't necessarily want to keep the data for 
+all the cells and genes
 
 #### Total counts as a quality measure
 
-One basic of method of exploring if a particular cell's data is trustworthy is
-seeing how many counts were measured in total. 
-Let's take a look at this data from how many counts are there per sample:
-Note that for `apply` the second argument indicates whether the function you 
-specify is being applied across all the rows or the columns (1 = rows, 2 = columns).
-Also note that in this case we are excluding the `genes` column for the calculation. 
+Let's take a look at this data from how many counts are there per sample. 
+We will use `apply` again here, but what will we use for the second argument?
 
 ```{r}
 # Make a vector of the total number of counts per sample
-total_counts <- apply(sc_data[, -1], 2, sum)
+total_counts <- apply(sc_data, 2, sum)
```

Comment:
Do you want to do
```suggestion
total_counts <- apply(sc_data, , sum)
```
instead? Or perhaps

```suggestion
total_counts <- apply(sc_data, <ROW OR COLUMN INDEX>, sum)
```

Reply:
What about this? I tried to tell R Markdown to `eval=FALSE` the chunk, but it still had problems with the <ROW OR COLUMN INDEX> portion. 


We will use `apply` again here, but what will we use for the second argument?
Copy and paste this code then replace `<ROW_OR_COLUMN_INDICATOR>` with the correct
argument. 
`total_counts <- apply(sc_data, <ROW_OR_COLUMN_INDICATOR>, sum)`

```{r}
# Make a vector of the total number of counts per sample using the above
# apply template
```

Reply:
Figured out an alternative. Because `total_counts` is used for a bunch of downstream steps, it makes it too complicated to do this here, instead, I changed it to have this 'fill in the blank' be for the `gene_means` step.